src/PostgREST/DbRequestBuilder.hs

```diff
@@ -46,13 +46,13 @@ readRequest maxRows allRels allProcs apiRequest  =
       let target = iTarget apiRequest in
       case target of
```

Comment:
For the relation detection to work the ```pdReturnType``` is needed, not the ```pdName```.
The pattern matching on ```pdReturnType``` is also needed since we only need Composite, it includes pg relations as documented in [pg_type](https://www.postgresql.org/docs/9.6/static/catalog-pg-type.html), not Scalars or Pseudo types.